docs/raw/serving-community-history.md

```diff
@@ -0,0 +1,315 @@
+- Messages to be exported MUST have a `timestamp` that lies within a given time range
+
+The range for the `timestamp` depends on the context in which the community owner node attempts to create a history archive. This can be one of the following:
+
+1. The community owner node attempts to create an archive periodically for the past seven days (including the current day). In this case, the `timestamp` has to lie within the day the last time an archive was created and the current day.
+2. The community owner node has been offline and attempts to create an archive for all the live messages it has missed since it went offline. In this case, the `timestamp` has to lie within the day the latest message was received and the current day.
+
+Exported messages MUST be restored as [14/WAKU2-MESSAGE](https://rfc.vac.dev/spec/14/) for bundling. Waku messages that have been exported for bundling can now be removed from the community owner node's database (community owner nodes still maintain a database of application messages).
+
+## Message history archives
+
+Message history archives are represented as `WakuMessageArchive` and created from messages exported from the local database. Message history archives are implemented using the following protocol buffer.
+
+### WakuMessageHistoryArchive
+
+The `from` field SHOULD contain a timestamp of the time range's lower bound.
+
+The `to` field SHOULD contain a timestamp of the time range's the higher bound.
+
+The `contentTopic` field MUST contain the same `contentTopic` that the archive's `messages` have.
+
+The `messages` field MUST contain all messages that belong into the archive given its `from`, `to` and `contentTopic` fields.
+
+```
+syntax = "proto3"
+
+message WakuMessageArchive {
+  uint64 from = 1
+  uint64 to = 2
+  string contentTopic = 3
+  repeated WakuMessage messages = 4 // `WakuMessage` is provided by 14/WAKU2-MESSAGE
+}
+```
+
+## Creating magnet links
+
+Once a message archive is created, the community owner node MUST derive a magnet link following the [Magnet URI scheme](https://en.wikipedia.org/wiki/Magnet_URI_scheme) using the underlying BitTorrent protocol client.
+
+The `dn` parameter ("display name") in the resulting magnet link MAY be optional.
+
+The resulting magnet link MUST be bundled into a `WakuMessageArchiveIndex`, which is then later distributed to other Status nodes.
+
+## Bundling history archives into archive indices
+
+Community owner nodes MUST provide message archives for the entire community history. However, each individual archive only contains a subset of the complete history, that is, either data for a time range of seven days, or, a time range in which the node was offline. Therefore, message history archives need to be bundled into a `WakuMessageArchiveIndex`, which later distributed via the Waku network and allows receiving nodes to fetch archives for individual time ranges.
```

Comment:
The entire history of each community (including all channels in that community) should be contained in a single binary blob.  Every time a new chunk of history is generated, it should be prepended to the preexisting binary blob, and a new magnet link created from this newly enlarged binary blog.  As such, members will only ever need to fetch the most recent magnet link from the hidden channel to access the entire history of a community.

Reply:
This is the tricky part. If we prepend this, community owners have to redownload the entire torrent every time as there's no easy way to recognise that some data of that torrent to be downloaded has already been downloaded (assuming that we can't control the size of `pieces`, which affects this). 

Tried to touch on this here: https://github.com/status-im/specs/pull/162#discussion_r767957001

Reply:
@PascalPrecht given that we *can* control the size of torrent pieces, as per our hangout discussion the other evening, this approach should be doable